include/LightGBM/config.h

```diff
@@ -356,6 +356,11 @@ struct Config {
   // desc = dropout rate: a fraction of previous trees to drop during the dropout
   double drop_rate = 0.1;
 
+  // alias = monotone_constraining_method
+  // desc = used only if ``monotone_constraints`` is set
+  // desc = monotone constraints method: if set to "basic", the most basic monotone constraints method will be used. It does not slow the library at all, but over-constrains the predictions. If set to "intermediate", a more advanced method will be used, which may slow the library very slightly. However, the intermediate method is much less constraining than the basic method and should significantly improve the results.
```

Comment:
@CharlesAuguste Have you published your paper from the [original comment](https://github.com/microsoft/LightGBM/pull/2305#issue-303471370) anywhere? If not yet, maybe host it at arXiv for now? I think that a link here to it will be very useful for users to understand the difference between methods.

Reply:
@guolinke @StrikerRUS I am working on uploading the report from the original comment somewhere. Right now we are thinking of arXiv as @StrikerRUS suggested. However, I need to get endorsement for that. I will be working on that but unless you have quick a solution for me, this may take some time. Anyway, I will upload the document and link it on all PR for reference as soon as possible. Thanks

Reply:
@CharlesAuguste Thank you very much! I believe, as a temporary solution (in)direct link to the original file will be just fine.
```
https://github.com/microsoft/LightGBM/files/3457826/PR-monotone-constraints-report.pdf
```

Reply:
Okay, I will edit my initial comment with the link for now. Thanks